File: python/정수삼각형(1932).py
n = int(input())
dp = [
    list(map(int,input().split()))
    for _ in range(n)
]

k = 2
for i in range(1, n):
    for j in range(k):
        if j == 0:
            dp[i][j] = dp[i][j] + dp[i-1][j]
        elif j == i:
            dp[i][j] = dp[i][j] + dp[i-1][j-1]
        else:
            dp[i][j] = max(dp[i-1][j-1], dp[i-1][j]) + dp[i][j]
    k += 1

print(max(dp[n-1]))

File: js/정수삼각형(1932).js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const n = Number(input[0]);
const dp = input.slice(1).map(line => line.split(' ').map(Number));

var k = 2;
for (let i=1; i<n; i++) {
    for (let j=0; j<k; j++) {
        if (j === 0) {
            dp[i][j] = dp[i][j] + dp[i-1][j];
        }
        else if (j === i) {
            dp[i][j] = dp[i][j] + dp[i-1][j-1];
        }
        else {
            dp[i][j] = Math.max(dp[i-1][j-1], dp[i-1][j]) + dp[i][j];
        }
    }
    k++;
}

console.log(Math.max(...dp[n-1]));